apps/widget-configurator/src/app/configurator/hooks/useSyncWidgetNetwork.ts

```diff
@@ -26,10 +28,10 @@ export function useSyncWidgetNetwork(
     }
   }, [isDisconnected, walletChainId, setNetworkControlState])
 
-  // Send a request to switch network if wallet network is different from widget network
+  // Send a request to switch network when user changes network in the configurator
   useEffect(() => {
-    if (!switchNetwork || isDisconnected || walletChainId === chainId) return
+    if (!switchNetwork || isDisconnected || walletChainIdRef.current === chainId) return
 
     switchNetwork(chainId)
-  }, [chainId, isDisconnected, switchNetwork, walletChainId, setNetworkControlState])
```

Comment:
The problem was here. Since `walletChainId` was in the deps we triggered network switchin in a loop